### PR TITLE
Add behaviour for no tags present

### LINKED ISSUE #82: Bug in error reporting
It seems that there is a bug related to exception handling. We are not checking the presence of the optional [here](https://github.com/shipkit/shipkit-auto-version/blob/04d9944424ed3f8e2fae4615032f47fa2da90a69/src/main/java/org/shipkit/auto/version/AutoVersion.java#L67). This causes build failures such as [those](https://github.com/mockito/mockito-kotlin/pull/435/checks?check_run_id=2623270165).

### LINKED ISSUE #83: Clean behavior when no tags present
Let's improve the behavior when no tags are present. When there are no tags (for example: it's a CI checkout with --no-tags, like [here](https://github.com/mockito/mockito-kotlin/runs/2623270165?check_suite_focus=true#step:2:40)) we should pick version `0.0.1-SNAPSHOT`.

### PR DESCRIPTION
**Pull request by shestee**
In AutoVersion in catch block the presence of the Optional was not checked; now it deducts version based on requestedVersion from config, and if it is not present `orElse()` returns fallback value for version: `0.0.1-SNAPSHOT`. This fixes #82 and also resolves #83 but additionally tag info finding is moved to `try-catch` block in `pickNextVersion()` - when an exception is caught fallback value for version is returned and more detailed explanation (than general one in AutoVersion) is logged. It is now also more visible in `NextVersionPicker` that `run("git", "describe", "--tags")` can throw an Exception.

### CODE REVIEW

File: src/main/java/org/shipkit/auto/version/NextVersionPicker.java [hunk 1]
```diff
@@ -33,8 +33,17 @@ String pickNextVersion(Optional<Version> previousVersion, VersionConfig config,
...
 
         if (!config.getRequestedVersion().isPresent()) {
-            String tag = runner.run("git", "describe", "--tags").trim();
+            String tag;
             String result;
+
+            try {
+                tag = runner.run("git", "describe", "--tags").trim();
+            } catch (Exception e) {
+                result = "0.0.1-SNAPSHOT";
```
Comment (koral--) [inline src/main/java/org/shipkit/auto/version/NextVersionPicker.java:42]: I think that exception should be somehow utilized. By logging it perhaps.

Comment (mockitoguy) [reply src/main/java/org/shipkit/auto/version/NextVersionPicker.java:42]: Good point. I think we should add both:
 - `info` level clean message
 - `debug` level message with stack trace

```
log.info("Process 'git describe --tags' exited with error. Assuming there are no tags. Run with --debug for more.");
log.debug("Ignored exception from 'git describe --tags'. Assuming there are no tags.", e);
```

@shestee, do you want to add those?

File: src/main/java/org/shipkit/auto/version/NextVersionPicker.java [hunk 2]
```diff
@@ -33,8 +33,17 @@ String pickNextVersion(Optional<Version> previousVersion, VersionConfig config,
...
+                explainVersion(log, result, "couldn't find tags in project");
```
Comment (mockitoguy) [inline src/main/java/org/shipkit/auto/version/NextVersionPicker.java]: "couldn't find tags in project" -> "couldn't run 'git describe --tags' (were the tags fetched?)"

**Review by mockitoguy — CHANGES_REQUESTED** [review]
I suggest a few changes and we should be ready to merge. Thanks!

File: src/main/java/org/shipkit/auto/version/NextVersionPicker.java (hunk 1 above)
Comment (shestee) [reply src/main/java/org/shipkit/auto/version/NextVersionPicker.java:42]: > I think that exception should be somehow utilized. By logging it perhaps.

Thanks for suggestion @koral-- ! You're right. I think logging the exception is fine here. I add `log.info()` and `log.debug()` as @mockitoguy proposed.

File: src/main/java/org/shipkit/auto/version/NextVersionPicker.java [hunk 3]
```diff
@@ -33,8 +33,20 @@ String pickNextVersion(Optional<Version> previousVersion, VersionConfig config,
...
             String result;
+
+            try {
+                tag = runner.run("git", "describe", "--tags").trim();
+            } catch (Exception e) {
+                result = "0.0.1-SNAPSHOT";
+                log.info("Process 'git describe --tags' exited with non-zero exit value. Assuming there are no tags. " +
+                        "Run with --debug for more.");
+                log.debug("Ignored exception from 'git describe --tags'. Assuming there are no tags.", e);
+                explainVersion(log, result, "couldn't run 'git describe --tags'");
```
Comment (mockitoguy) [inline src/main/java/org/shipkit/auto/version/NextVersionPicker.java]: "couldn't run 'git describe --tags'" -> "couldn't run 'git describe --tags' (assuming there are no tags)"

**Review by mockitoguy — CHANGES_REQUESTED** [review]
One little change and we're ready to merge!

File: src/main/java/org/shipkit/auto/version/NextVersionPicker.java (hunk 3 above)
Comment (shestee) [reply src/main/java/org/shipkit/auto/version/NextVersionPicker.java]: @mockitoguy thanks, it's done

**Pull request merged**